packages/story-editor/src/components/canvas/editElement.js

```diff
@@ -17,80 +17,62 @@
 /**
  * External dependencies
  */
+import { useState, forwardRef } from '@googleforcreators/react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
 import { useUnits } from '@googleforcreators/units';
 
 /**
  * Internal dependencies
  */
-import { useState, useRef } from '@googleforcreators/react';
 import { getDefinitionForType } from '../../elements';
 import {
   elementWithPosition,
   elementWithSize,
   elementWithRotation,
 } from '../../elements/shared';
-import SingleSelectionMoveable from './singleSelectionMoveable';
 
 const Wrapper = styled.div`
   ${elementWithPosition}
   ${elementWithSize}
-	${elementWithRotation}
-	pointer-events: initial;
+  ${elementWithRotation}
+  pointer-events: initial;
 `;
 
-function EditElement({ element }) {
+const EditElement = forwardRef(function EditElement(
+  { element, editWrapper, onResize },
+  ref
+) {
   const { id, type } = element;
   const { getBox } = useUnits((state) => ({
     getBox: state.actions.getBox,
   }));
 
-  const [editWrapper, setEditWrapper] = useState(null);
   // Needed for elements that can scale in edit mode.
   const [localProperties, setLocalProperties] = useState(null);
 
-  const { Edit, hasEditModeMoveable } = getDefinitionForType(type);
+  const { Edit } = getDefinitionForType(type);
   const box = getBox(
     localProperties ? { ...element, ...localProperties } : element
   );
```

Comment:
Just wondering why not:
```js
  const [localProperties, setLocalProperties] = useState({});
  const box = getBox({ ...element, ...localProperties }); // in both places
```

Reply:
Yeah, you can even spread `null`, that does break. I'll change this.

Reply:
Hmm, but spreading and re-structuring the (property-rich) element without cause is pretty wasteful actually. But I will move it to a separate variable.

Reply:
Reworked it a bit in 4a44d28bb.

Reply:
Yes and no :smile: 1000000 element spread operations took ~54ms, while simple assignment took ~0.35ms, quite a big difference - yes, does 0.000054 vs 0.00000035 do any real-world difference - no, given that it is executed only in edit mode, on pre/post resize text element and such, because on every media edit (scale, move) you will have those localProperties and it will spread.

Either way, 👍🏻   for moving that to a variable.

Reply:
Well, that depends if spreads are re-written in babel to pre-es5 support or not. I actually do not know. I kind of assumed it was, and if so, it is definitely more expensive. But if it's transpiled, we should be good.